### ISSUE TITLE
teardown reports "Error" when a half-installed cluster is destoyed

### ISSUE THREAD

**Issue by nehaberry** [red-hat-storage/ocs-ci#325]
Description:

There are scenarios when "deployment" of a cluster fails due to some resource constraints(VPC limit, load balancer limit) , etc.

In the event of above failure, if we initiate teardown of the said cluster, even though teardown finally succeeds, the run fails with following error message:

```
09:22:19 - MainThread - ocs_ci.ocs.openshift_ops - ERROR - Cluster is not ready to use: Error during execution of command: ['oc', 'cluster-info'].
                                                           Error is Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host
FAILED         

```

But as the OC cluster was not up , its expected  not to get any output for "cluster-info" command. Hence, we should handle this and not fail the test run with "--teardown"

**Full output of failure**

```
(new-ocs-ci) [nberry@localhost ocs-ci]$ run-ci  -m deployment  --cluster-name nberry-jul2-1 --cluster-path /home/nberry/aws-install/jul2-1/ --log-level=DEBUG --teardown
CMD error:: Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host

Cluster is not ready to use: Error during execution of command: ['oc', 'cluster-info'].
Error is Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host

Collecting Cluster versions
CMD error:: Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host

========================================== test session starts ==========================================
platform linux -- Python 3.7.0, pytest-4.6.3, py-1.8.0, pluggy-0.12.0
rootdir: /home/nberry/ocs-ci-scripts-master/ocs-ci-scripts/ocs-ci, inifile: pytest.ini, testpaths: tests
plugins: reportportal-1.0.5, html-1.21.1, metadata-1.8.0
collected 28 items / 27 deselected / 1 selected                                                         

tests/ecosystem/deployment/test_ocs_basic_install.py::test_cluster_is_running 
-------------------------------------------- live log setup ---------------------------------------------
09:22:18 - MainThread - root - INFO - RUN ID: 1562039537
09:22:18 - MainThread - root - INFO - All logs are located under /tmp/logs
09:22:18 - MainThread - tests.conftest - INFO - Running OCS basic installation
09:22:18 - MainThread - tests.conftest - INFO - Will teardown cluster because --teardown was provided
09:22:18 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/nberry/aws-install/jul2-1/auth/kubeconfig
09:22:18 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
09:22:19 - MainThread - ocs_ci.utility.utils - ERROR - CMD error:: Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host
09:22:19 - MainThread - ocs_ci.ocs.openshift_ops - ERROR - Cluster is not ready to use: Error during execution of command: ['oc', 'cluster-info'].
                                                           Error is Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host
09:22:19 - MainThread - tests.conftest - INFO - Attempting teardown of non-accessible cluster: /home/nberry/aws-install/jul2-1/
--------------------------------------------- live log call ---------------------------------------------
09:22:19 - MainThread - ocs_ci.ocs.openshift_ops - INFO - Testing access to cluster with /home/nberry/aws-install/jul2-1/auth/kubeconfig
09:22:19 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc cluster-info
09:22:19 - MainThread - ocs_ci.utility.utils - ERROR - CMD error:: Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host
09:22:19 - MainThread - ocs_ci.ocs.openshift_ops - ERROR - Cluster is not ready to use: Error during execution of command: ['oc', 'cluster-info'].
                                                           Error is Unable to connect to the server: dial tcp: lookup api.nberry-jul2-1.qe.rh-ocs.com on 10.75.5.25:53: no such host
FAILED         


                                                        []
09:26:02 - MainThread - tests.conftest - INFO - Destroying the test cluster complete
=============================== 1 failed, 27 deselected in 223.89 seconds ===============================
(new-ocs-ci) [nberry@localhost ocs-ci]$ 


```

Please suggest.

**Comment by nehaberry** [comment]
```
=============================================== FAILURES ================================================
________________________________________ test_cluster_is_running ________________________________________

    @deployment
    def test_cluster_is_running():
>       assert is_cluster_running(config.ENV_DATA['cluster_path'])
E       AssertionError: assert False
E        +  where False = is_cluster_running('/home/nberry/aws-install/jul2-1/')

tests/ecosystem/deployment/test_ocs_basic_install.py:12: AssertionError

```

**Comment by nehaberry** [comment]
@petr-balogh @vavuthu please suggest .

**Comment by vavuthu** [comment]
Assigning to myself

**Comment by clacroix12** [comment]
Was this fully addressed by #377 ?

**Comment by vasukulkarni** [comment]
Closing stale issue.
